R/topics.R

```diff
@@ -1,21 +1,35 @@
 # @return An integer vector giving selected topics
-select_topics <- function(match_strings, topics) {
+select_topics <- function(match_strings, topics, check = FALSE) {
   n <- nrow(topics)
   if (length(match_strings) == 0) {
     return(integer())
   }
 
   indexes <- purrr::map(match_strings, match_eval, env = match_env(topics))
 
+  if (length(purrr::keep(indexes, ~ length(.x) > 0)) == 0 && check) {
+    topic_must(
```

Comment:
When is this triggered? The ordering seems weird to me given that we do the +/- topic selection below.

Reply:
This is triggered if a) one or more topics is specified in _pkgdown.yml and b) none of the specified topics have a match. Perhaps this is more of a corner case.